src/components/EmojiPicker/EmojiPicker.js

```diff
@@ -10,166 +10,143 @@ const DEFAULT_ANCHOR_ORIGIN = {
+     * @param {Function} [onEmojiSelectedValue=() => {}] - Run a callback when Emoji selected.
+     * @param {Element} emojiPopoverAnchorValue - Element to which Popover is anchored
+     * @param {Object} [anchorOrigin=DEFAULT_ANCHOR_ORIGIN] - Anchor origin for Popover
+     * @param {Function} [onWillShow=() => {}] - Run a callback when Popover will show
+     */
+    function showEmojiPicker(onModalHideValue, onEmojiSelectedValue, emojiPopoverAnchorValue, anchorOrigin, onWillShow = () => {}) {
+        setOnModalHide(() => onModalHideValue);
+        setOnEmojiSelected(() => onEmojiSelectedValue);
+        setEmojiPopoverAnchor(emojiPopoverAnchorValue);
 
-    componentDidUpdate(prevProps, prevState) {
-        if (prevState.isEmojiPickerVisible === this.state.isEmojiPickerVisible || !this.state.isEmojiPickerVisible) {
-            return;
+        if (emojiPopoverAnchorValue) {
+            // Drop focus to avoid blue focus ring.
+            emojiPopoverAnchorValue.blur();
         }
 
-        // Dismiss the keyboard to provide a focus for the emoji picker to avoid selection issues.
-        Keyboard.dismiss();
-    }
-
-    componentWillUnmount() {
-        if (!this.emojiPopoverDimensionListener) {
-            return;
-        }
-        this.emojiPopoverDimensionListener.remove();
+        measureEmojiPopoverAnchorPosition(emojiPopoverAnchorValue).then((value) => {
+            onWillShow();
+            setIsEmojiPickerVisible(true);
+            setEmojiPopoverAnchorPosition(value);
+            setEmojiPopoverAnchorOrigin(anchorOrigin || DEFAULT_ANCHOR_ORIGIN);
+        });
     }
 
-    /**
-     * Callback for the emoji picker to add whatever emoji is chosen into the main input
-     *
-     * @param {String} emoji
-     * @param {Object} emojiObject
-     */
-    selectEmoji(emoji, emojiObject) {
-        // Prevent fast click / multiple emoji selection;
-        // The first click will hide the emoji picker by calling the hideEmojiPicker() function
-        // and in that function the emojiPopoverAnchor prop to will be set to null (synchronously)
-        // thus we rely on that prop to prevent fast click / multiple emoji selection
-        if (!this.emojiPopoverAnchor) {
-            return;
+    useEffect(() => {
+        if (isEmojiPickerVisible) {
+            Keyboard.dismiss();
         }
 
-        this.hideEmojiPicker();
-        if (_.isFunction(this.onEmojiSelected)) {
-            this.onEmojiSelected(emoji, emojiObject);
-        }
-    }
+        const emojiPopoverDimensionListener = Dimensions.addEventListener('change', () => {
+            measureEmojiPopoverAnchorPosition(emojiPopoverAnchor).then((value) => {
+                setEmojiPopoverAnchorPosition(value);
+            });
+        });
+        return () => {
+            emojiPopoverDimensionListener.remove();
+        };
+    }, [isEmojiPickerVisible, emojiPopoverAnchor]);
```

Comment:
if `emojiPopoverAnchor` is a ref, there is no need to add it as a dependency, because refs don't produce rerenders, so if the value of `emojiPopoverArchor.current` changes, `useEffect` won't get executed